components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2419,48 +2487,84 @@ void KeyringService::HasPendingUnlockRequest(
   std::move(callback).Run(HasPendingUnlockRequest());
 }
 
-absl::optional<std::string> KeyringService::GetBitcoinReceivingAddress(
-    const std::string& keyring_id,
-    uint32_t account_index,
-    uint32_t receiving_index) {
-  if (keyring_id != mojom::kBitcoinKeyringId) {
-    return absl::nullopt;
-  }
+std::vector<std::pair<std::string, mojom::BitcoinKeyIdPtr>>
+KeyringService::GetBitcoinAddresses(const std::string& keyring_id,
```

Comment:
Consider using some new class like BitcoinAccount which allows to a list of derived addrs by depth